satpy/resample.py

```diff
@@ -823,11 +823,144 @@ def compute(self, data, expand=True, **kwargs):
+                                                source_lons,
+                                                source_lats)
+
+    def compute(self, data, **kwargs):
+        """Call the resampling."""
+        raise NotImplementedError("Use the sub-classes")
+
+    def resample(self, data, **kwargs):
+        """Resample `data` by calling `precompute` and `compute` methods.
+
+        Args:
+            data (xarray.DataArray): Data to be resampled
+
+        Returns (xarray.DataArray): Data resampled to the target area
+        """
+        self.precompute(**kwargs)
+        attrs = data.attrs.copy()
+        data_arr = data.data
+        if data.ndim == 3 and data.dims[0] == 'bands':
+            dims = ('bands', 'y', 'x')
+        # Both one and two dimensional input data results in 2D output
+        elif data.ndim in (1, 2):
+            dims = ('y', 'x')
+        else:
+            dims = data.dims
+
+        result = self.compute(data_arr, **kwargs)
+        if result.ndim > len(dims):
+            result = da.squeeze(result)
+        result = xr.DataArray(result, dims=dims, coords=data.coords,
+                              attrs=attrs)
+        return result
+
+
+class BucketAvg(BucketResamplerBase):
+    """Class for averaging bucket resampling.
+
+    Bucket resampling calculates the average of all the values that
+    are closest to each bin and inside the target area.
+
+    Parameters
+    ----------
+    fill_value : float (default: np.nan)
+        Fill value for missing data
+    mask_all_nans : boolean (default: False)
+        Mask all locations with all-NaN values
+    """
+
+    def compute(self, data, fill_value=np.nan, mask_all_nan=False, **kwargs):
+        """Call the resampling."""
+        results = []
+        if data.ndim == 3:
+            for i in range(data.shape[0]):
+                res = self.resampler.get_average(data[i, :, :],
+                                                 fill_value=fill_value,
+                                                 mask_all_nan=mask_all_nan)
+                results.append(res)
+        else:
+            res = self.resampler.get_average(data, fill_value=fill_value,
+                                             mask_all_nan=mask_all_nan)
+            results.append(res)
+
+        return da.stack(results)
+
+
+class BucketSum(BucketResamplerBase):
+    """Class for bucket resampling which implements accumulation (sum).
+
+    This resampler calculates the cumulative sum of all the values
+    that are closest to each bin and inside the target area.
+
+    Parameters
+    ----------
+    fill_value : float (default: np.nan)
+        Fill value for missing data
+    mask_all_nans : boolean (default: False)
+        Mask all locations with all-NaN values
+    """
+
+    def compute(self, data, mask_all_nan=False, **kwargs):
+        """Call the resampling."""
+        LOG.debug("Resampling %s", str(data.name))
+        results = []
+        if data.ndim == 3:
```

Comment:
Is it possible to do this on the `y` and `x` dimensions instead of handling "banded" data? I don't have an example off-hand but if we do things on dimension then it will be more flexible in the future. I am also ok delaying this functionality if it turns out to be a time suck.

Similarly, is it possible to call the resampler method with xarray handling the splitting and stacking? Like "Xarray call this function for each element along dimension X".

Reply:
Here we are handling pure `dask` arrays without `xarray` envelope, so don't have coordinates/dims associated with the data. Making all the steps to use `xarray` objects instead will need another PR.